docs/sql-ref-syntax-qry-select-hints.md

```diff
@@ -126,6 +126,10 @@ Join hints allow users to suggest the join strategy that Spark should use. Prior
 
     Suggests that Spark use shuffle-and-replicate nested loop join.
 
+* **BROADCAST_NL**
+
+    Suggests that Spark use broadcast nested loop join. The join side with the hint will be broadcast regardless of `autoBroadcastJoinThreshold`. If both sides of the join have the broadcast hints, the one with the smaller size (based on stats) will be broadcast.
```

Comment:
should we highlight that this hint is useful when the join type is not supported by hash join? and list these join types, such as full outer join

Reply:
yea, updated the docs